File: src/main/java/hdzi/editstarters/dependency/Dependency.java
package hdzi.editstarters.dependency;

/**
 * 依赖
 */
public interface Dependency extends Point {
    String getGroupId();

    String getArtifactId();

    @Override
    default String point() {
        return getGroupId() + ":" + getArtifactId();
    }
}

File: src/main/java/hdzi/editstarters/springboot/bean/Dependency.kt
package hdzi.editstarters.springboot.bean

/**
 * Created by taojinhou on 2019/1/16.
 */
interface Dependency {
    val groupId: String?
    val artifactId: String?

    val point: String
        get() = "${groupId}:${artifactId}"
}